woocommerce/woocommerce
# Migrate 'Parent Category' field to SelectTree

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This migrates the 'Parent Category' field of the Create Category Modal to the SelectTree component, similar to the Category field.

Some tweaks had to be done on the component to accommodate this new scenario (single selection), and also because it's inside a modal.

Closes #37633 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the new product manager
2. Go to the "Categories" field
3. Press "Create new" (the create new category modal should open)
4. Create a new category and fill the parent category using only the mouse.
5. Save and check if it saved correctly
6. Create a new category and fill the parent category using only the keyboard by typing an existing category.
7. Save and check if it saved correctly
8. Create a new category and fill the parent category using only the keyboard using the up, down, home, end keys. Select one using space.
9. Save and check if it saved correctly
10. Perform any additional smoke tests you'd like.
